share-through-htb.js

```diff
@@ -201,6 +145,30 @@ function ShareThroughHtb(configs) {
         System.documentWrite(doc, adm);
     }
 
+    function __generateAdm(bid, placementKey) {
+        var stxResponseName = "str_response_" + bid.bidId;
+        var bidJsonString = JSON.stringify(bid);
+
+        return "<div data-str-native-key='" + placementKey + "' data-stx-response-name='" + stxResponseName + "'></div>" +
+        "<script>var " + stxResponseName + " = " + bidJsonString + " </script>" +
+        "<script src='//native.sharethrough.com/assets/sfp-set-targeting.js'></script>" +
```

Comment:
What is the purpose of this file?

Reply:
We insert a script tag with src=`//native.sharethrough.com/assets/sfp.js` onto the parent page (lines 157 - 168). `sfp.js` sets up a message listener and contains our rendering code.

Assuming this ad markup is being served into an iframe, `sfp-set-targeting.js` is a file that gathers relevant data and posts the data to the message listener set up by `sfp.js`.

When that message listener gets the creative data, `sfp.js` will render the ad in place of the iframe.

Reply:
This script is not async, can we make this async? 

If we are understanding correctly, this will be the creative/adm that gets put on the page. Does this mean it will load for each creative that wins. This could result in multiple instances of the script loading.

Reply:
Yes we're expecting to load it sync, as we think it is small enough not to interrupt the user experience, and we don't want the browser to delay downloading the file too much. 

`sfp-set-targeting.js` is a ~2 KB script that loads into each iframe (when we have a winning creative) and communicates its ad payload with the one message listener on the main page (which is set up by `sfp.js`).

To answer the question: yes we're expecting an instance of the script to download for each winning creative.